src/Ombi.Core/Engine/BaseMediaEngine.cs

```diff
@@ -78,6 +78,26 @@ public int ResultLimit
             return _dbTv;
         }
 
+        protected async Task<RequestEngineResult> CheckCanManageRequest(BaseRequest request) {
+            
+            // Admins can always manage requests
+            var isAdmin = await IsInRole(OmbiRoles.PowerUser) || await IsInRole(OmbiRoles.Admin);
+            if (isAdmin)
+                return new RequestEngineResult { Result = true };
+
+            // Users with 'ManageOwnRequests' can only manage their own requests
```

Comment:
We could do this check first as it's potentially quicker than retrieving the user from the DB

something like 
```

var canManageOwnRequests = await IsInRole(OmbiRoles.ManageOwnRequests);
if (!canManageOwnRequests)
{
    return new RequestEngineResult
    {
        Result = false,
        ErrorCode = ErrorCode.NoPermissions
    };
}
// Users with 'ManageOwnRequests' can only manage their own requests
var isRequestedBySameUser = (await GetUser().Id).Equals(request.RequestedUser.Id);
if (isRequestedBySameUser)
{
    return new RequestEngineResult { Result = true };
}
```

Reply:
Makes sense, but FYI IsInRole also calls GetUser() so it will just prevent calling it twice for nothing. Maybe this method could be improved but it's out of scope of this PR.